### DISCUSSION TITLE
Problem with dynamically changing src for model-viewer

### DISCUSSION THREAD

**Discussion by PCPbiscuit** [google/model-viewer#4308] [Q&A]
### Description

I'm having somewhat of a problem trying to recrate the first example from https://modelviewer.dev/examples/augmentedreality/. 
My slider works perfectly well on WEB and not so perfect on mobile. 
When I open it in AR viewer on mobile it shows the model but once I click on any of the slides, the model disappears. If I close and reopen it again, it shows the model which I clicked

I'm using React for this project, however I'm simply linking script in index.html's head and not using any external libs to handle model-viewer. 
Here is some code for reference: 

 ```
               <model-viewer
                  id='test'
                  ar
                  // ar-modes='scene-viewer webxr'
                  camera-controls
                  class='w-full min-h-10 md:hidden'
                  src={models[currentModelIndex]?.model}
                  reveal='manual'
                  alt='A 3D shoes'
                >
                  <button
                    slot='ar-button'
                    style={{
                      backgroundColor: 'black',
                      color: 'white',
                      borderRadius: '4px',
                      border: 'none',
                      position: 'absolute',
                      top: '0px',
                      left: '0',
                    }}
                  >
                    Activate AR
                  </button>
                  <div
                    className='w-full absolute bottom-0 overflow-hidden'
                    id='slider'
                  >
                    <div className='flex overflow-x-auto snap-mandatory scroll-smooth'>
                      {models.map((model, index) => (
                        <button
                          className='flex border-none mr-2 bg-cover bg-no-repeat w-24 h-24 rounded-lg shrink-0 bg-center'
                          onClick={handleModel(index)}
                          style={{ backgroundImage: `url(${model.preview})` }}
                          key={index}
                        ></button>
                      ))}
                    </div>
                  </div>
                </model-viewer>
```

### OS
- [+] Android
- [+] Linux
- [+] Windows

I'm hoping somebody would direct me to some docs\related issues on why this isn't working
If my description isn't clear enough, I could try to recreate it in CodeSandbox

**Comment by elalish** [comment]
There's a lot of things that could be going wrong, and it's hard to say without having your URL to see. Have you checked with simpler models? It could be that you have some that are far from the origin or scaled oddly. Are there errors or warnings in your JS console?

**Reply by PCPbiscuit** [reply]
Everything works perfectly fine until I click any of the slides on mobile. 

All models are pretty simple (I thought about them too so I found simplest ones on the web just for tests), they work and they load quickly, they scale well etc.
There are no console errors on PC, I have no idea how to open console log on mobile

Would it be easier if I try and recreate this in CodeSandbox?

**Reply by elalish** [reply]
Yes, a sandbox would be ideal! If you're on Android, go to chrome://inspect on your laptop, plug in your phone and you can open Chrome's debugger on any page your Android's Chrome has open.

**Reply by PCPbiscuit** [reply]
Nice! I'll try to debug on my phone and if it doesn't help I'll link codesandbox here

**Reply by PCPbiscuit** [reply]
For some reason I couldn't even manage to show slider in codesandbox..

However I deployed the same code to Vercel: https://codesandbox-3js-test.vercel.app/
Code repository: https://github.com/PCPbiscuit/codesandbox-3js-test

As you can see, it works perfectly fine until you try to use slider in mobile AR
I'm using android phone

**Reply by PCPbiscuit** [reply]
I appreciate any kind of feedback, not asking for a working solution or anything. A hint or a link in the  documentation would be sufficient

**Reply by PCPbiscuit** [reply]
I haven't changed anything and it works now...

**Reply by diegoteran** [reply]
Glad it works now! we didn't have time to look into this before, but if it happens again feel free to send us a live demo and we could try to help
